packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -360,21 +530,22 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
                 _proof: _withdrawalProof,
                 _root: _outputRootProof.messagePasserStorageRoot
             }) == false
-        ) revert InvalidMerkleProof();
+        ) {
+            revert OptimismPortal_InvalidMerkleProof();
+        }
 
-        // Designate the withdrawalHash as proven by storing the `disputeGameProxy` & `timestamp` in the
-        // `provenWithdrawals` mapping. A `withdrawalHash` can only be proven once unless the dispute game it proved
-        // against resolves against the favor of the root claim.
+        // Designate the withdrawalHash as proven by storing the disputeGameProxy and timestamp in
+        // the provenWithdrawals mapping. A given user may re-prove a withdrawalHash multiple
+        // times, but each proof will reset the proof timer.
         provenWithdrawals[withdrawalHash][msg.sender] =
-            ProvenWithdrawal({ disputeGameProxy: gameProxy, timestamp: uint64(block.timestamp) });
-
-        // Emit a `WithdrawalProven` event.
-        emit WithdrawalProven(withdrawalHash, _tx.sender, _tx.target);
-        // Emit a `WithdrawalProvenExtension1` event.
-        emit WithdrawalProvenExtension1(withdrawalHash, msg.sender);
+            ProvenWithdrawal({ disputeGameProxy: _disputeGameProxy, timestamp: uint64(block.timestamp) });
 
         // Add the proof submitter to the list of proof submitters for this withdrawal hash.
         proofSubmitters[withdrawalHash].push(msg.sender);
+
+        // Emit a WithdrawalProven events.
+        emit WithdrawalProven(withdrawalHash, _tx.sender, _tx.target);
+        emit WithdrawalProvenExtension1(withdrawalHash, msg.sender);
```

Comment:
Do we still need both of these events? It's a nit but wondering if/when we can clean up these duplicates?

Reply:
We might be able to remove them but probably too late for this PR, resolving